objects/c932e8/script.lua

```diff
@@ -13,42 +40,110 @@ function onLoad()
+        end
+    end
+    return false
+end
+
 function doSpiritSetup(params)
     local color = params.color
     if not Global.getVar("gameStarted") then
         Player[color].broadcast("Please wait for the game to start before pressing button!", "Red")
         return
     end
 
-    local zone = getObjectFromGUID(Global.getVar("elementScanZones")[color])
-    local objs = zone.getObjects()
-    local found = false
-    local serpent = nil
-    for _,obj in pairs(objs) do
-        if obj.guid == "cebe09" then
-            found = true
-            serpent = obj
-            break
-       end
-    end
-    if not found then
+    if not findSerpent(color) then
         Player[color].broadcast("You have not picked Serpent Slumbering Beneath the Island!", "Red")
         return
     end
 
     local snapPoints = serpent.getSnapPoints()
-
-    table.insert(snapPoints,{position={1.14, 0.20, 0.65}})
-    table.insert(snapPoints,{position={0.92, 0.20, 0.65}})
-    table.insert(snapPoints,{position={0.70, 0.20, 0.65}})
-    table.insert(snapPoints,{position={1.25, 0.20, 0.84}})
-    table.insert(snapPoints,{position={1.03, 0.20, 0.84}})
-    table.insert(snapPoints,{position={0.81, 0.20, 0.84}})
+    for _, position in pairs(slumberPoints) do
+        table.insert(snapPoints, {position = position})
+    end
     serpent.setSnapPoints(snapPoints)
 
-    self.destruct()
+    playerBlock = Global.getVar("playerBlocks")[color]
+
+    initialized = true
+    self.clearButtons()
+    self.locked = true
+    self.interactable = false
+    local position = self.getPosition()
+    position.y = -2
+    self.setPosition(position)
+
+    updatePresence()
+    Wait.time(updatePresence, 1, -1)
+end
+
+local function presenceOnIsland(color)
+    local count = 0
+    for _, presence in pairs(getObjectsWithTag("Presence")) do
+        if presence.getName() == color .. "'s Presence" then
+            local hits = Physics.cast{
+                origin = presence.getBoundsNormalized().center,
+                size = presence.getBoundsNormalized().size,
+                direction = Vector(0, -1, 0),
+                max_distance = 1,
+                type = 2, -- box
+            }
+            for _, hit in pairs(hits) do
+                local obj = hit.hit_object
+                if obj.hasTag("Balanced") or obj.hasTag("Thematic") then
+                    local quantity = presence.getQuantity()
+                    if quantity == -1 then
+                        count = count + 1
+                    else
+                        count = count + quantity
+                    end
+                    break
+                end
+            end
+        end
+    end
+    return count
+end
+
+local function slumberPresence()
+    local count = 0
+    for _, position in pairs(slumberPoints) do
+        local hits = Physics.cast{
+            origin = serpent.positionToWorld(position), -- pos
+            direction = Vector(0, 1, 0),
+            max_distance = 1,
```

Comment:
max_distance could probably be a smaller value right?

Reply:
Maybe? I'm not sure there is reason to make it smaller, though.